### PR TITLE
ODIM-64 Add events conf in odimra config

### PR DESCRIPTION
**Pull request by irappa-pattar**

### CODE REVIEW

File: odim-controller/helmcharts/odimra-config/templates/configmaps.yaml [hunk 1]
```diff
@@ -129,6 +129,11 @@ data:
...
     		"TelemetryService"
     	],
-        "ConnectionMethodConf": {{ .Values.odimra.connectionMethodConf | toJson }},
-    	"SupportedPluginTypes": ["Compute", "Fabric", "Storage"]
+      "ConnectionMethodConf": {{ .Values.odimra.connectionMethodConf | toJson }},
+    	"SupportedPluginTypes": ["Compute", "Fabric", "Storage"],
+      "EventConf": {
+                 "DeliveryRetryAttempts" : 3,
+                 "DeliveryRetryIntervalSeconds" : 60,
+                 "RetentionEventsInMinutes" : 10
```
Comment (jeevan-kamkar) [inline odim-controller/helmcharts/odimra-config/templates/configmaps.yaml]: Rename it to `RetentionOfUndeliveredEventsInMinutes`

File: lib-utilities/config/odimra_config.json [hunk 2]
```diff
@@ -128,5 +128,10 @@
 		  "ConnectionMethodType": "Redfish",
 		  "ConnectionMethodVariant":"Compute:BasicAuth:URP_v1.0.0"
 	  }
-   ]
+  ],
+  "EventConf": {
+		"DeliveryRetryAttempts" : 3,
+		"DeliveryRetryIntervalSeconds" : 60,
+		"RetentionEventsInMinutes" : 10
```
Comment (jeevan-kamkar) [inline lib-utilities/config/odimra_config.json]: Rename it to `RetentionOfUndeliveredEventsInMinutes`

File: lib-utilities/config/constants.go [hunk 3]
```diff
@@ -90,6 +93,12 @@ const (
 	DefaultTLSMaxVersion = tls.VersionTLS12
 	// DefaultTLSServerVerify - indicator for performing server validation
 	DefaultTLSServerVerify = true
+	// DefaultDeliveryRetryAttempts - default DeliveryRetryAttempts value
+	DefaultDeliveryRetryAttempts = 3
+	// DefaultDeliveryRetryIntervalSeconds - default DeliveryRetryIntervalSeconds value
+	DefaultDeliveryRetryIntervalSeconds = 60
+	// DefaultRetentionEventsInMinutes - default RetentionEventsInMinutes value
+	DefaultRetentionEventsInMinutes = 10
```
Comment (jeevan-kamkar) [inline lib-utilities/config/constants.go]: Rename it to `DefaultRetentionOfUndeliveredEventsInMinutes`

File: lib-utilities/config/config.go [hunk 4]
```diff
@@ -151,6 +152,13 @@ type ConnectionMethodConf struct {
 	ConnectionMethodVariant string `json:"ConnectionMethodVariant"`
 }
 
+// EventConf stores all inforamtion related to event delivery configurations
+type EventConf struct {
+	DeliveryRetryAttempts        int `json:"DeliveryRetryAttempts"`        // holds value of retrying event posting to destination
+	DeliveryRetryIntervalSeconds int `json:"DeliveryRetryIntervalSeconds"` // holds value of retrying events posting in interval
+	RetentionEventsInMinutes     int `json:"RetentionEventsInMinutes"`     // holds value of how long we can retain the events
```
Comment (jeevan-kamkar) [inline lib-utilities/config/config.go]: Rename it to `RetentionOfUndeliveredEventsInMinutes`

**Review by jeevan-kamkar — CHANGES_REQUESTED** [review]
Address the requested changes

File: odim-controller/helmcharts/odimra-config/templates/configmaps.yaml (hunk 1 above)
Comment (irappa-pattar) [reply odim-controller/helmcharts/odimra-config/templates/configmaps.yaml]: changed

File: lib-utilities/config/odimra_config.json (hunk 2 above)
Comment (irappa-pattar) [reply lib-utilities/config/odimra_config.json]: changed

File: lib-utilities/config/constants.go (hunk 3 above)
Comment (irappa-pattar) [reply lib-utilities/config/constants.go]: changed

File: lib-utilities/config/config.go (hunk 4 above)
Comment (irappa-pattar) [reply lib-utilities/config/config.go]: changed

**Review by jeevan-kamkar — APPROVED** [review]

File: lib-utilities/config/config.go [hunk 5]
```diff
@@ -531,3 +542,34 @@ func checkConnectionMethodConf() error {
...
+		Data.EventConf.DeliveryRetryAttempts = DefaultDeliveryRetryAttempts
+	}
+	if Data.EventConf.DeliveryRetryIntervalSeconds <= 0 {
+		log.Warn("No value found for DeliveryRetryIntervalSeconds, setting default value")
+		Data.EventConf.DeliveryRetryIntervalSeconds = DefaultDeliveryRetryIntervalSeconds
+	}
+	if SaveUndeliveredEventsFlag {
+		reattempt := Data.EventConf.DeliveryRetryAttempts * Data.EventConf.DeliveryRetryIntervalSeconds
+		if reattempt < (Data.EventConf.RetentionOfUndeliveredEventsInMinutes / 60) {
+			return fmt.Errorf("configured value can't be less than the delivery attempt values")
```
Comment (shivacharanms) [inline lib-utilities/config/config.go]: Mention which configured value cant be less

Comment (irappa-pattar) [reply lib-utilities/config/config.go]: updated

**Review by shivacharanms — APPROVED** [review]

**Pull request merged**